pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/startree/AggregationFunctionColumnPair.java

```diff
@@ -66,6 +66,19 @@ public static AggregationFunctionColumnPair fromAggregationConfig(StarTreeAggreg
     return fromFunctionAndColumnName(aggregationConfig.getAggregationFunction(), aggregationConfig.getColumnName());
   }
 
+  /**
+   * Return a new {@code AggregationFunctionColumnPair} from an existing functionColumnPair where the new pair
+   * has the {@link AggregationFunctionType} set to the aggregated function type used in the segment or indexes.
+   * @param functionColumnPair the existing functionColumnPair
+   * @return the new functionColumnPair
+   */
+  public static AggregationFunctionColumnPair resolveToAggregatedType(
```

Comment:
Suggest renaming to `resolveToStoredType()` which IMO is more clear

Reply:
I agree, that name is better.  Updated in https://github.com/apache/pinot/pull/12164/commits/2ca0cf588a484cf40c8ec57cb8c482b24f72ce71